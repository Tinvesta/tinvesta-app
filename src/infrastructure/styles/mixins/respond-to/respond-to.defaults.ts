import { TBreakpoints } from 'use-device-detect';

import {
  LG_SCREEN_BREAKPOINT,
  MD_SCREEN_BREAKPOINT,
  SM_SCREEN_BREAKPOINT,
  XL_SCREEN_BREAKPOINT,
  XS_SCREEN_BREAKPOINT,
} from '@constants';

export const deviceBreakpoint: TBreakpoints = {
  xl: XL_SCREEN_BREAKPOINT,
  lg: LG_SCREEN_BREAKPOINT,
  md: MD_SCREEN_BREAKPOINT,
  sm: SM_SCREEN_BREAKPOINT,
  xs: XS_SCREEN_BREAKPOINT,
};
