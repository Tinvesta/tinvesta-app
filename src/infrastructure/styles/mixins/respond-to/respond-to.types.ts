import { css } from '@emotion/react';

export type TCSSParams = Parameters<typeof css>;
export type TMediaFor = 'min-width' | 'max-width';
