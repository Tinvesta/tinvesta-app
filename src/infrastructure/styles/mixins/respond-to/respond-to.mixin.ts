import { css } from '@emotion/react';
import { objectKeys } from 'ts-object-keys';

import { deviceBreakpoint } from './respond-to.defaults';
import { TCSSParams, TMediaFor } from './respond-to.types';

const keys = objectKeys(deviceBreakpoint);

const respondToDecorator = (mediaFor: TMediaFor) =>
  keys.reduce((accumulator, label) => {
    accumulator[label] = (...args: TCSSParams) => css`
      @media (${mediaFor}: ${deviceBreakpoint[label]}px) {
        ${css(...args)};
      }
    `;

    return accumulator;
  }, {} as Record<keyof typeof deviceBreakpoint, Function>);

export const respondToMin = respondToDecorator('min-width');

export const respondToMax = respondToDecorator('max-width');
