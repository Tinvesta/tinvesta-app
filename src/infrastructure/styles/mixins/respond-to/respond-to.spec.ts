import { functionImportTest, toMatchSnapshot } from '@utils';

import { respondToMax, respondToMin } from './respond-to.mixin';

describe('respondToMin mixin', () => {
  functionImportTest(respondToMin.xl);
  toMatchSnapshot(() => respondToMin.xl`background-color: red;`);
});

describe('respondToMax mixin', () => {
  functionImportTest(respondToMax.xl);
  toMatchSnapshot(() => respondToMax.xl`background-color: red;`);
});
