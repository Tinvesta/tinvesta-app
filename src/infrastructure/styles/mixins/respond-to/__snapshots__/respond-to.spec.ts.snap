// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`respondToMax mixin snapshot test should match snapshot 1`] = `
Object {
  "map": undefined,
  "name": "xg27cz",
  "next": undefined,
  "styles": "
      @media (max-width: 1600px) {
        background-color: red;;;
      }
    ",
  "toString": [Function],
}
`;

exports[`respondToMin mixin snapshot test should match snapshot 1`] = `
Object {
  "map": undefined,
  "name": "r1m74f",
  "next": undefined,
  "styles": "
      @media (min-width: 1600px) {
        background-color: red;;;
      }
    ",
  "toString": [Function],
}
`;
